### PR TITLE
Update TeleporterMessenger.sol

### PR DESCRIPTION
**Pull request by Aryan9592**
## Why this should be merged
This PR aims to enhance some 'for' loops by making them more gas efficient. Also got some typos fixed!

## How this works
For reference -> https://gist.github.com/grGred/9bab8b9bad0cd42fc23d4e31e7347144#for-loops-improvement

Thanks @michaelkaplan13

### CODE REVIEW

File: contracts/src/Teleporter/TeleporterMessenger.sol [hunk 1]
```diff
@@ -583,9 +590,12 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
         }
 
         // Otherwise, the deliverer address must be included in allowedRelayers.
-        for (uint256 i = 0; i < allowedRelayers.length; ++i) {
+        for (uint256 i; i < allowedRelayers.length;) {
             if (allowedRelayers[i] == delivererAddress) {
                 return true;
+                unchecked {
```
Comment (bernard-avalabs) [inline contracts/src/Teleporter/TeleporterMessenger.sol:596]: I assume that this unchecked increment should be outside of the if block.

Comment (Aryan9592) [reply contracts/src/Teleporter/TeleporterMessenger.sol]: @bernard-avalabs  You are right about it..Thanks!

File: contracts/src/Teleporter/TeleporterMessenger.sol [hunk 2]
```diff
@@ -269,7 +269,7 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
             "TeleporterMessenger: invalid destination chain ID"
         );
 
-        // Check the message has not been delivered previously.
+        // Check if the message has not been delivered previously.
```
Comment (geoff-vball) [inline contracts/src/Teleporter/TeleporterMessenger.sol:272]:
```suggestion
        // Require that the message has not been delivered previously.
```

File: contracts/src/Teleporter/TeleporterMessenger.sol [hunk 3]
```diff
@@ -303,14 +303,17 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
...
-        for (uint256 i = 0; i < length; ++i) {
+        for (uint256 i; i < length;) {
             TeleporterMessageReceipt memory receipt = teleporterMessage
                 .receipts[i];
             _markReceipt(
                 warpMessage.sourceChainID,
                 receipt.receivedMessageID,
                 receipt.relayerRewardAddress
             );
+            unchecked {
```
Comment (minghinmatthewlam) [inline contracts/src/Teleporter/TeleporterMessenger.sol:314]: can you add a comment about why we're using unchecked increment here, same for below loop. Thinking if the minimal gas optimization is worth the readability.

Comment (Aryan9592) [reply contracts/src/Teleporter/TeleporterMessenger.sol:314]: Sure!

Comment (Aryan9592) [reply contracts/src/Teleporter/TeleporterMessenger.sol:317]: As you say @minghinmatthewlam

**Review by michaelkaplan13 — APPROVED** [review]
🙏 Thanks for catching all of these and making the PR. LGTM

**Review by geoff-vball — APPROVED** [review]

**Comment by michaelkaplan13** [comment]
I think the Snyk action is failing just because the forked branch isn't able to use the Snyk account key set for the repository. It should pass when merged now.

**Comment by minghinmatthewlam** [comment]
Thanks @Aryan9592 for the fixes!

**Comment by Aryan9592** [comment]
Welcome @minghinmatthewlam

**Pull request merged**